.devcontainer/generate_compile_commands

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+BUILD=0
+
+HELP_MESSAGE="
+Usage: ./generate_compile_commands [OPTIONS]
+
+Generate compile commands for clangd. The following options are supported:
+
+  --build                  Build the project to also generate protobuf files.
+  -h|--help                Show this help message.
+"
+
+while [ "$#" -gt 0 ]; do
+  case "$1" in
+    --build) BUILD=1; shift 1;;
+    -h|--help) echo "$HELP_MESSAGE"; exit 0;;
+  esac
+done
+
+sudo mkdir /ws && sudo ln -s /src /ws/src
+sudo chown ubuntu:ubuntu /ws
+cd /ws && mkdir build
+cmake -B build -S src -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DBUILD_TESTING=ON -DBUILD_PROTOCOL=ON
+if [ $BUILD -eq 1 ]; then
+  cmake --build build
```

Comment:
@domire8, this is my only concern. This is single core invocation and takes a long time. Should we perhaps do
`
``suggestion
  cmake --build build -j $(( `nproc` > 1 ? `nproc` - 1 : num_cores ))
```

Reply:
that would give max number of jobs - 1 to hog all resources

Reply:
Sure you're suggestion makes sense, I didn't pay attention how long it takes. Compared to the time to build all the dependencies it cannot be that bad right? :smile:

Reply:
no it's not comparable to that, but still 😛 ...